src/ServiceControl/MessageRedirects/Api/MessageRedirectsModule.cs

```diff
@@ -0,0 +1,144 @@
+﻿namespace ServiceControl.MessageRedirects.Api
+{
+    using System;
+    using System.Linq;
+    using Nancy;
+    using NServiceBus;
+    using Raven.Client;
+    using ServiceBus.Management.Infrastructure.Extensions;
+    using ServiceBus.Management.Infrastructure.Nancy.Modules;
+    using ServiceControl.Infrastructure;
+    using ServiceControl.Infrastructure.Extensions;
+    using ServiceControl.MessageRedirects.InternalMessages;
+
+    public class MessageRedirectsModule : BaseModule
+    {
+        public IBus Bus { get; set; }
+
+        public MessageRedirectsModule()
+        {
+
+            Post["/redirects"] = parameters =>
+            {
+                var message = new CreateMessageRedirect
+                {
+                    FromPhysicalAddress = parameters.fromphysicaladdress,
+                    ToPhysicalAddress = parameters.tophysicaladdress
+                };
+
+                if (string.IsNullOrWhiteSpace(message.FromPhysicalAddress) || string.IsNullOrWhiteSpace(message.ToPhysicalAddress))
+                {
+                    return HttpStatusCode.BadRequest;
+                }
+
+                message.MessageRedirectId = DeterministicGuid.MakeId(message.FromPhysicalAddress, message.ToPhysicalAddress);
+
+                using (var session = Store.OpenSession())
+                {
+                    var redirect = session.Load<MessageRedirect>(MessageRedirect.GetDocumentIdFromMessageRedirectId(message.MessageRedirectId));
+
+                    if (redirect != null)
+                    {
+                        return Negotiate.WithReasonPhrase("Duplicate").WithModel(redirect).WithStatusCode(HttpStatusCode.Conflict);
+                    }
+
+                    var dependents = session.Query<MessageRedirect>().Where(r => r.ToPhysicalAddress == message.FromPhysicalAddress).ToList();
+
+                    if (dependents.Any())
```

Comment:
Actually I thought more about this and still think this is the only check that is needed. The opposite scenario is excluded from this validation is where multiple queues are redirected to the same queue, which should be supported.